packages/flame/lib/src/game/mixins/has_parent.dart

```diff
@@ -0,0 +1,13 @@
+import '../../../components.dart';
+
+/// A mixin that ensures a parent is of the given type [T].
+mixin HasParent<T extends Component> on Component {
+  @override
+  T get parent => super.parent! as T;
+
+  @override
+  void onMount() {
+    assert(super.parent is T, 'Parent must be of type ${T.toString()}');
+    super.onMount();
+  }
```

Comment:
```suggestion
  @override
  Future<void> addToParent(Component parent) async {
    assert(
      parent is T,
      'Parent must be of type ${T.toString()}.',
    );
    await super.addToParent(parent);
  }
```

Reply:
1. `'$T'` is simpler than `${T.toString()}`
2. `addToParent()` is not async

Reply:
Agree with 1, just copied the same assert message and didn't pay attention. Now thinking about it, both approaches will display an invalid string if toString is overrided.

With 2 that's true. But wouldn't the computation be stopped before in addToParent than in onMount?

Reply:
Good call for doing this on the addToParent! Will test if that works as expected but I am quite sure it will

Also the `toString` is a Github copilot "mistake", will turn that into `$T`. The toString can't be overridden by the user as it is a generic Type so I don't think that is a concern for us.

Reply:
Reworked it to the `addToParent`, works as expected.